src/controllers/verify-yc.ts

```diff
@@ -0,0 +1,134 @@
+import { Body, Controller, Ctx, Post } from 'amala'
+import { Context } from 'vm'
+import { badRequest } from '@hapi/boom'
+import { ethers, utils } from 'ethers'
+import { polygonProvider } from '@/helpers/providers'
+import AddressVerifyBody from '@/validators/AddressVerifyBody'
+import BalanceUniqueVerifyBody from '@/validators/BalanceUniqueVerifyBody'
+import EmailUniqueVerifyBody from '@/validators/EmailUniqueVerifyBody'
+import TwitterBody from '@/validators/TwitterBody'
+import eddsaSigPoseidon from '@/helpers/signatures/eddsaSigPoseidon'
+import fetchUserProfile from '@/helpers/twitter/fetchUserProfile'
+import getBalance from '@/helpers/getBalance'
+import poseidonHash from '@/helpers/signatures/poseidonHash'
+import sendEmail from '@/helpers/sendEmail'
+import zeroAddress from '@/models/zeroAddress'
+
+@Controller('/verify-yc')
+export default class VerifyYCController {
+  @Post('/email-unique')
+  async sendUniqueEmail(
+    @Body({ required: true }) { email }: EmailUniqueVerifyBody
+  ) {
+    // 1. Create `emailHash = poseidon([0, email])`, (0 = email attestation)
+    const eddsaMessage = ['0', utils.hexlify(utils.toUtf8Bytes(email))]
+
+    const emailHash = await poseidonHash(eddsaMessage)
+
+    // 2. Sign `[0, emailHash]` with EdDSA, (0 = yc attestation)
+    const eddsaSignature = await eddsaSigPoseidon([
+      0,
+      ...utils.toUtf8Bytes(emailHash),
+    ])
+
+    const domain = email.split('@')[1].toLowerCase()
+
+    // TODO:
+    // 3. Send email with signature (must look nice and ketl branded)
+    void sendEmail({
```

Comment:
Feels like we should extract procedures before into separate helper function like `encryptAndSignEmail`

Reply:
The controller should be as simple as possible